source/developing.html.erb.md

```diff
@@ -20,7 +20,7 @@ These introductory tips cover the basics of accessible front-end web development
 {:.attach_permalink}
 ## Associate all form elements with labels
 
-Associate the `<label>` element with form elements using the *for* and *id* attributes. Position the `<label>` element offscreen using <abbr>CSS</abbr> when the design does not include a visible label. Beware of doing this as many sighted people may still misinterpret the required input without an obvious label.
+Associate every form control with a `<label>` element, ideally using the *for* and *id* attributes. In specific situations it may be acceptable to hide `<label>` elements visually, but in most cases labels are needed to help readers understand the required input.
```

Comment:
I am wondering how to keep this one as simple as possible. For example, every form element need not have an `<label>` element, as ARIA allows for different forms. Agreed that off-screen CSS is now ideally handled using a `aria-label` attribute. I wouldn't say it is outdated, as I still see plenty of use. Suggest something like:

> Associate every form control with a label. This can be achieved in different ways, such as using linked `for` and `id` attributes, `aria-labeledby` attribute, or `aria-label` attribute. Use `aria-label` when a design does not provide for a visible label, although use this with caution as most labels are needed to help understand what input is required.